app/controllers/agent/ManageAgentController.scala

```diff
@@ -0,0 +1,132 @@
+                                         searchParams = GetPropertyLinksParameters(),
+                                         pagination = PaginationParams(
+                                           startPoint = 1,
+                                           pageSize = 1000,
+                                           requestTotalRowCount = false),
+                                         representationStatusFilter =
+                                           Seq(RepresentationApproved, RepresentationPending)
+                                       )
+    } yield {
+      (propertyLinks.total, agentToBeManagedOpt) match {
+        case (0, Some(agent)) if agent.propertyCount == 0 =>
+          //IP has no property links but still has an agent
+          Some(removeAgentFromOrganisation(submitManageAgentForm, agent))
+        case (1, Some(agent)) if agent.propertyCount == 0 =>
+          //IP has one property link but agent is not assigned
+          Some(
+            manageAgentPage(submitManageAgentForm, ManageAgentOptions.onePropertyLinkNoAssignedAgentsOptions, agent))
+        case (1, Some(agent)) if agent.propertyCount == 1 =>
+          //IP has one property link and agent is assigned to that property
+          Some(unassignAgentFromProperty(submitManageAgentForm, agent))
+        case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks > 1 && agent.propertyCount == 0 =>
+          //IP has more than one property links but agent is not assigned to any
+          Some(
+            manageAgentPage(
+              submitManageAgentForm,
+              ManageAgentOptions.multiplePropertyLinksNoAssignedAgentsOptions,
+              agent))
+        case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks > agent.propertyCount =>
+          //agent is not assigned to all of the IP's property links
+          Some(
+            manageAgentPage(
+              submitManageAgentForm,
+              ManageAgentOptions.multiplePropertyLinksAgentAssignedToSomeOptions,
+              agent))
+        case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks == agent.propertyCount =>
+          //agent is assigned to all of the IP's property links
+          Some(
+            manageAgentPage(
+              submitManageAgentForm,
+              ManageAgentOptions.multiplePropertyLinksAgentAssignedToAllOptions,
+              agent))
+        case _ => None
+      }
+    }
+
+  def submitManageAgent(agentCode: Long): Action[AnyContent] = authenticated.async { implicit request =>
+    ManageAgentRequest.submitManageAgentRequest.bindFromRequest.fold(
+      errors => {
+        getManageAgentPage(Some(agentCode), errors).map {
+          case None       => NotFound(errorHandler.notFoundTemplate)
+          case Some(page) => BadRequest(page)
+        }
+      }, { success =>
+        ??? //fixme this will be implemented in subsequent stories
```

Comment:
Could return 501 Not Implemented
Any Jira number for the story that will implement this?